lib/html-proofer/runner.rb

```diff
@@ -27,11 +27,14 @@ def initialize(src, opts = {})
           @options[:url_swap][Regexp.new("^//#{dom}")] = ''
         end
       end
-
-      @failures = []
     end
 
     def run
+      @stat = init_stat
+      if @options[:stat]
+        @stat.print_header
+      end
+
       @logger.log :info, "Running #{checks} on #{@src} on *#{@options[:extension]}... \n\n"
```

Comment:
Since we removed `return if @hide` above, this area seems like a good place to refactor:

``` ruby
if @options[:stat]
  @state = init_stat
  @stat.print_header
else
  @logger.log :info, "Running ..."
end
```

and so on. Basically flip on the logging iff `@stat` is `false`.